### PR TITLE
Add example code to tf.data.Dataset.filter() documentation

### PR DESCRIPTION
**Pull request by kweinmeister**

### CODE REVIEW

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
Hi, 

Thanks for the change.

I know a lot of them are already like this, but anything you can do to make the example more copy-paste runnable helps.

File: tensorflow/python/data/ops/dataset_ops.py [hunk 1]
```diff
@@ -1100,6 +1100,21 @@ def interleave(self,
   def filter(self, predicate):
     """Filters this dataset according to `predicate`.
 
+    ```python
+    # NOTE: The following examples use `{ ... }` to represent the
+    # contents of a dataset.
+    a = { 1, 2, 3 }
+    b = { 4, 5, 6, 7 }
+
+    a.filter(lambda x: x < 3) == { 1, 2 }
```
[the remark this answers is not in the document]
Comment (kweinmeister) [reply tensorflow/python/data/ops/dataset_ops.py]: I changed the expected result to be commented out. Please let me know if any other changes need to be made after the latest commit.

File: tensorflow/python/data/ops/dataset_ops.py [hunk 2]
```diff
@@ -1100,6 +1100,21 @@ def interleave(self,
   def filter(self, predicate):
     """Filters this dataset according to `predicate`.
 
+    ```python
+    # NOTE: The following examples use `{ ... }` to represent the
+    # contents of a dataset.
+    a = { 1, 2, 3 }
+    b = { 4, 5, 6, 7 }
```
[the remark this answers is not in the document]
Comment (kweinmeister) [reply tensorflow/python/data/ops/dataset_ops.py]: Done!

**Review by MarkDaoust — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**